cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -337,30 +337,49 @@ std::pair<VersionedItem, FrameAndDescriptor> LocalVersionedEngine::read_datafram
     return std::make_pair(version.value_or(VersionedItem{}), std::move(frame_and_descriptor));
 }
 
-std::pair<VersionedItem, std::optional<google::protobuf::Any>> LocalVersionedEngine::read_descriptor_version_internal(
-        const StreamId& stream_id,
-        const VersionQuery& version_query
+folly::Future<std::pair<VersionedItem, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_descriptor(
+    std::optional<AtomKey>&& key,
+    const StreamId& stream_id,
+    const VersionQuery& version_query){
+    missing_data::check<ErrorCode::E_NO_SUCH_VERSION>(static_cast<bool>(key.has_value()),
+        "Unable to retrieve descriptor data. {}@{}: version not found", stream_id, version_query);
```

Comment:
I think the semantic of passing an optional when you're only going to throw is a bit weird. If I see a function signature that takes an optional, that implies that it can handle either the thing being present or it being absent. If the parameter is required, then it's up to the caller to check

Reply:
i.e.. the purpose of an optional parameter is to indicate that a thing is indeed optional. If it's not optional then don't use it

Reply:
Seconded